src/record_layout.zig

```diff
@@ -0,0 +1,459 @@
+
+        computeLayout(fld.ty, comp, &type_layout);
+
+        // The required alignment of the field is the maximum of the required alignment of the
+        // underlying type and the __declspec(align) annotation on the field itself.
+        // See test case 0028.
+        var req_align = type_layout.required_alignment_bits;
+        if (annotationAlignment(fld_attrs)) |anno| {
+            req_align = std.math.max(anno, req_align);
+        }
+
+        // The required alignment of a record is the maximum of the required alignments of its
+        // fields except that the required alignment of bitfields is ignored.
+        // See test case 0029.
+        if (fld.bit_width == null) {
+            self.req_align_bits = std.math.max(self.req_align_bits, req_align);
+        }
+
+        // The offset of the field is based on the field alignment of the underlying type.
+        // See test case 0027.
+        var fld_align_bits = type_layout.field_alignment_bits;
+        if (self.max_field_align_bits) |max_align| {
+            fld_align_bits = std.math.min(fld_align_bits, max_align);
+        }
+
+        if (isPacked(fld_attrs)) {
+            // __attribute__((packed)) on a field is a clang extension. It behaves as if #pragma
+            // pack(1) had been applied only to this field. See test case 0057.
+            fld_align_bits = BITS_PER_BYTE;
+        }
+        // __attribute__((packed)) on a field is a clang extension. It behaves as if #pragma
+        // pack(1) had been applied only to this field. See test case 0057.
+        fld_align_bits = std.math.max(fld_align_bits, req_align);
+        if (fld.bit_width) |bit_width| {
+            self.layoutBitField(type_layout.size_bits, fld_align_bits, fld.name_tok != 0, bit_width);
+        } else {
+            self.layoutRegluarField(type_layout.size_bits, fld_align_bits, fld);
+        }
+    }
+    fn layoutBitField(self: *MscvContext, size_bits: u64, field_align: u64, is_named: bool, bit_width: u32) void {
+        _ = self;
+        _ = size_bits;
+        _ = field_align;
+        _ = is_named;
+        _ = bit_width;
+        // TODO
+    }
+
+    fn layoutRegluarField(self: *MscvContext, size_bits: u64, field_align: u32, fld: *Field) void {
+        self.contains_non_bitfield = true;
+        self.ongoing_bitfield = null;
+        self.pointer_align_bits = std.math.max(self.pointer_align_bits, field_align);
+        self.field_align_bits = std.math.max(self.field_align_bits, field_align);
+        const offset_bits = switch (self.is_union) {
+            true => 0,
+            false => std.mem.alignForwardGeneric(u64, self.size_bits, field_align),
+        };
+        self.size_bits = std.math.max(self.size_bits, offset_bits + size_bits);
+        fld.layout = .{
+            .offset_bits = offset_bits,
+            .size_bits = size_bits,
+        };
+    }
+};
+
+pub fn recordLayout(ty: *Type, comp: *const Compilation, parser: ?*const Parser) void {
```

Comment:
Is `parser.pragma_pack` the only value we ever need from the parser? Would it be simpler to just pass in a `?u8` pragma_pack value instead of the whole parser?

Reply:
yeah. I used to pull other things out of the parser, but as you and Vexu implemented attributes, I guess just the pragma is left.